kani-compiler/src/codegen_cprover_gotoc/compiler_interface.rs

```diff
@@ -115,6 +115,13 @@ impl GotocCodegenBackend {
             call_graph,
         );
 
+        // Re-collect reachable items after global transformations were applied. This is necessary
```

Comment:
I think this would require re-implementing reachability traversal inside the global pass, or at least something very similar to it. Another advantage of running reachability before is that reachability provides global passes with a call graph. It is not used now, but it could help resolving function pointer and vtable calls if we want to add support for that in the future.

Reply:
Ah, I see. Please add a comment that mentions this advantage.